component/attr_cache/attr_cache.go

```diff
@@ -521,7 +521,7 @@ func NewAttrCacheComponent() internal.Component {
 // On init register this component to pipeline and supply your constructor
 func init() {
 	internal.AddComponent(compName, NewAttrCacheComponent)
-	attrCacheTimeout := config.AddInt32Flag("attr-cache-timeout", int32(defaultAttrCacheTimeout), "attribute cache timeout")
+	attrCacheTimeout := config.AddUint32Flag("attr-cache-timeout", 0, "attribute cache timeout")
```

Comment:
If IsSet works do we need to change the default value here ? If user has not given anything then let it be to default value. Putting 0 here will fail the IsSet right ?

Reply:
This is the only way the is set works. I've tested this and if there is no timeout in attr cache or cli it will use 120

Reply:
I think I have some misunderstanding here. So when we specify the cli-flags we also mention a default value. So ideally if user has not provided anything in cli options then code trying to get its value shall get the default value. If it does not populate the value to default then what is the point of defining default value here? If we are saying not giving this option on cli yeilds 0 then in most cases it will be 0 only. If we are adding attr-cache in pipeline it means user wants to cache but then value 0 will just contradict it. As long as this component is there in pipeline value 0 shall be the last value that we expect. If user misses to provide the value that is where we want the default value.

Reply:
unfortunately if you look at the code in keys_tree it uses if the value is 0 or not to determine if it was set. So technically the CLI function will not work if we pass in --cli=0 -> this will bump it to default

Reply:
this is also wrong. User can set file-cache-timeout or fuse timeouts to 0 through cli. So if its not working that way then lets fix that as well

Reply:
"not provided", "zero value", "non zero value" shall be three different state and all possible combination of those shall be supported.